programs/ziti-edge-tunnel/netif_driver/windows/tun.c

```diff
@@ -472,3 +475,22 @@ int set_dns(netif_handle tun, uint32_t dns_ip) {
 char* get_tun_name(netif_handle tun) {
     return tun->name;
 }
+
+static BOOL CALLBACK
+tun_delete_cb(_In_ WINTUN_ADAPTER_HANDLE adapter, _In_ LPARAM param) {
+    wchar_t name[32];
+    WintunGetAdapterName(adapter, name);
+    wchar_t tun_name[32] = ZITI_TUN;
```

Comment:
you could avoid hard-coding the adapter name here if you passed `tun` as a parameter to cleanup_adapters and then via the param argument to WintunEnumAdapters.